File: CarRental/02-BO/BranchModel.cs
using System.ComponentModel.DataAnnotations;

namespace _02_BO
{
    public class BranchModel
    {

        [Required]
        [StringLength(100,MinimumLength =7)]
        public string Address { get; set; }

        [Required]
        [StringLength(20, MinimumLength = 3)]
        public string BranchName { get; set; }

        [Required]
        public GeographicCoordinates Coordinates { get; set; }

        public int BranchId { get; set; }
    }
}

File: car-rental-angular/src/app/shared/models/branch.model.ts
export class Branch {
    Address:string;
    BranchName:string;
    Coordinates:GeographicCoordinates;
    BranchId:number;
    /**
     *
     */
    constructor() {
        this.Coordinates= new GeographicCoordinates();
    }
}
export class GeographicCoordinates {

    Latitude:number;
    LatitudeMinutes :number;
    IsNorth :boolean=true;
    Longitude :number;
    LongitudeMinutes :number;
    IsEast :boolean=true;
    toString ():string
     {
        return `${this.Latitude}°${this.LatitudeMinutes}'${this.IsNorth?'N':'S'} ${this.Longitude}°${this.LongitudeMinutes}'${this.IsEast?'E':'W'}`;
      }
}
